Client/Frontend/Browser/TabManager.swift

```diff
@@ -232,30 +232,70 @@ class TabManager : NSObject {
 }
 
 extension TabManager {
+    class SavedTab: NSObject, NSCoding {
+        let url: NSURL?
+        let isReaderMode: Bool
+        let isSelected: Bool
+
+        init(url: NSURL, isReaderMode: Bool, isSelected: Bool) {
+            self.url = url
+            self.isReaderMode = isReaderMode
+            self.isSelected = isSelected
+        }
+
+        required init(coder: NSCoder) {
+            self.url = coder.decodeObjectForKey("url") as? NSURL
+            self.isReaderMode = coder.decodeBoolForKey("isReaderMode")
+            self.isSelected = coder.decodeBoolForKey("isSelected")
+        }
+
+        func encodeWithCoder(coder: NSCoder) {
+            coder.encodeObject(url, forKey: "url")
+            coder.encodeBool(isReaderMode, forKey: "isReaderMode")
+            coder.encodeBool(isReaderMode, forKey: "isSelected")
+        }
+    }
+
     func encodeRestorableStateWithCoder(coder: NSCoder) {
-        coder.encodeInteger(count, forKey: "tabCount")
-        coder.encodeInteger(selectedIndex, forKey: "selectedIndex")
-        for i in 0..<count {
-            let tab = tabs[i]
-            coder.encodeObject(tab.url!, forKey: "tab-\(i)-url")
+        var savedTabs = [SavedTab]()
+        for (tabIndex, tab) in enumerate(tabs) {
+            if let url = tab.url, displayURL = tab.displayURL {
```

Comment:
The point of app serialization/deserialization is to freeze/restore the state as close as possible to the state before the app was killed, so if the user had about:home tabs, it's more accurate to restore them than to forget them.

Since we only restore the active URL on iOS and not the rest of the tab history, it's not as big of a deal for us as it would be on Android or desktop. There is at least one case where not restoring about:home tabs feels jarring, though: when an about:home tab was selected before the app was killed. Since the tab was selected, but we don't save it, that means we'll arbitrarily select a different tab (the first tab) when we restore.